Automattic/jetpack
# Dependencies: Fix broken prod dependencies

#### Changes proposed in this Pull Request:

- Drop production-only dependency installation. There's very limited benefit to skipping `devDependency` installation (save some time in an infrequent process) and it leads to issues that _only_ occur when making the production bundle.
- Add all [upgrade-nudge dependencies](https://github.com/Automattic/jetpack/blob/190301ba02a13ef148513348279e1f579f13a927/extensions/shared/components/upgrade-nudge/index.jsx#L4-L11) to dependencies. With upgrade-nudge SSR (https://github.com/Automattic/jetpack/pull/13070), these dependencies will be required.

This was working because `@wordpress/url` is installed as a transitive dependency, `@wordpress/e2e-test-utils` (a devDependency) causes it to be installed. If `@wordpress/e2e-test-utils` (like with `yarn install --production`) - `@wordpress/url` is not installed.

I believe, this is only a problem with extensions that have "SSR" from the implementation in https://github.com/Automattic/jetpack/pull/13070, at the moment only Simple Payments. At runtime in the browser, `@wordpress/*` dependencies are _externals_, meaning they are pulled from the runtime environment. That makes the presence or absence of a `@wordpress/*` package irrelevant. However, with https://github.com/Automattic/jetpack/pull/13070 SSR, some of these dependencies will be required _at build time_.

See p1566905517230900-slack-luna / d871cf9983b11e37dea0e1f70009b4ec84571857

#### Testing instructions:
* `yarn build-production` builds without errors.
* Add a Simple Payments block to a site on a free plan. The upgrade nudge should be visible in the editor and on the frontend for admin users (see https://github.com/Automattic/jetpack/pull/13070)
